### PR TITLE
cadl support

### PR DESCRIPTION
**Pull request by dw511214992**

### CODE REVIEW

File: tools/sdk-generation-pipeline/documents/docker/README.md [hunk 1]
```diff
@@ -19,11 +19,14 @@ The docker image will be used in different scenarios:
...
-docker run -it --privileged  -v {local_spec_repo_path}:/spec-repo -v {local_work_folder}:/work-dir -v {local_autorest_config}:/autorest.md sdkgeneration.azurecr.io/sdk-generation:latest --readme={relative_readme} --sdk={sdk_to_generate}
-```
-
+1. Generate codes with Autorest:
+    ```shell
+    docker run -it --privileged -v {local_spec_repo_path}:/spec-repo -v {local_work_folder}:/work-dir -v {local_autorest_config}:/autorest.md sdkgeneration.azurecr.io/sdk-generation:latest --readme={relative_readme} --sdk={sdk_to_generate}
+    ```
+2. Generate codes with CADL:
+    ```shell
+    docker run -it --privileged -v {local_spec_repo_path}:/spec-repo -v {local_work_folder}:/work-dir -v sdkgeneration.azurecr.io/sdk-generation:latest --readme={relative_cadl_project} --sdk={sdk_to_generate}
```
Comment (zzvswxy) [inline tools/sdk-generation-pipeline/documents/docker/README.md:28]: if should replace --readme with --cadl-project

Comment (dw511214992) [reply tools/sdk-generation-pipeline/documents/docker/README.md]: yes. updated.

**Review by zzvswxy — DISMISSED** [review]
ok for me, but nice to have another one to reviewed

File: tools/sdk-generation-pipeline/pipelines/steps/generate-sdk.yml [hunk 2]
```diff
@@ -33,33 +36,42 @@ steps:
     condition: ne('${{ parameters.triggerType }}', 'release')
     
   - bash: |
-      if [ "${{ parameters.serviceType }}" = "resource-manager" ];then
+      if [ !  -z "${{ parameters.cadlProject }}" ]; then
```
Comment (zzvswxy) [inline tools/sdk-generation-pipeline/pipelines/steps/generate-sdk.yml:39]: `  - name: relativeCadlProjectPath
    type: string
    default: "''"
`

${{ parameters.cadlProject }} = ''
`if [ ! -z "${{ parameters.cadlProject }}" ]`
may the judgement function always be  true

File: tools/sdk-generation-pipeline/pipelines/steps/generate-sdk.yml [hunk 3]
```diff
@@ -5,6 +5,9 @@ parameters:
   - name: cadlProject
     type: string
 
+  - name: cadlProject
```
Comment (zzvswxy) [inline tools/sdk-generation-pipeline/pipelines/steps/generate-sdk.yml:8]: same with line 5?

**Pull request merged**